lib/msgpack.rb

```diff
@@ -1,6 +1,13 @@
 require "msgpack/version"
-begin
-  require "msgpack/#{RUBY_VERSION[/\d+.\d+/]}/msgpack"
-rescue LoadError
-  require "msgpack/msgpack"
+
+if defined?(RUBY_ENGINE) && RUBY_ENGINE == 'jruby'
+  require 'jruby'
```

Comment:
There was a specific reason for `require 'jruby'`: the line below uses `JRuby.runtime`, which is defined in `jruby.rb`. I think neither `require 'jruby'` nor `require 'java'` is needed in newer JRuby versions (1.7+), they're required by default, that's why it appears to work just as well with `require 'java'`.